corehq/apps/app_manager/models.py

```diff
@@ -1233,6 +1233,14 @@ class MappingItem(DocumentSchema):
     # lang => localized string
     value = DictProperty()
 
+    @property
+    def escaped_key(self):
+        """
+        Return the key with spaces replaces by underscores for use in xml
+        variable names
+        """
+        return self.key.replace(" ", "_")
```

Comment:
I think a better solution is instead of having an `escaped_key` function, you should have a `key_as_variable` function. That should return `'k{key}'.format(key=key)` if that would result in a valid variable name (i.e. if there are no spaces), and, like, `'h{hash}'.format(hash=hashlib.md5(key).hexdigest()[:8])` otherwise. Good things about this scheme:

- we don't arbitrarily treat `' '` and `'_'` as the same character
- it strictly extends current practice (as did the method you were using) and so in most cases will be readable
- it is extensible to other characters beyond `' '` should that be necessary